src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -1827,13 +2090,87 @@ instruct vsubD(vReg dst, vReg src1, vReg src2) %{
   ins_pipe(pipe_slow);
 %}
 
+// vector sub - predicated
+
+instruct vsubB_masked(vReg dst_src1, vReg src2, vRegMask_V0 vmask) %{
+  match(Set dst_src1 (SubVB (Binary dst_src1 src2) vmask));
+  ins_cost(VEC_COST);
+  format %{ "vsub.vv $dst_src1, $src2, $vmask\t#@vsubB_masked" %}
+  ins_encode %{
+    __ rvv_vsetvli(T_BYTE, Matcher::vector_length_in_bytes(this));
+    __ vsub_vv(as_VectorRegister($dst_src1$$reg), as_VectorRegister($dst_src1$$reg),
+               as_VectorRegister($src2$$reg), Assembler::v0_t);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vsubS_masked(vReg dst_src1, vReg src2, vRegMask_V0 vmask) %{
+  match(Set dst_src1 (SubVS (Binary dst_src1 src2) vmask));
```

Comment:
Can we just merge those match rules in one instruct just like `vlsrIL`? Looks like those instructs only differ from BasicType.

Reply:
Fixed.